WordPress/gutenberg
# Navigation Editor: Move selected menu ID to the store

## Description
Attempt to move currently selected menu ID to the store. Value is synced to localStorage as suggested here https://github.com/WordPress/gutenberg/pull/31132#issuecomment-827353433.

PR introduces new actions/selector for working with selected menu ID.

### Other changes
* Adds `useSelectedMenuId` hook for convenience.
* Removes `useSelectedMenuData` hook and `MenuIdContext`.

Fixes #22623.

## How has this been tested?
1. Got to the new experimental Navigation screen.
2. Create few menus.
3. Select the last menu created.
4. Refreshing page should display the same menu.

Testing the store changes:

`npm run test-unit -- packages/edit-navigation/src/store/test`

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
